runner/launcher/src/mill/launcher/MillLauncherMain.java

```diff
@@ -33,14 +36,16 @@ public static void main(String[] args) throws Exception {
       }
     }
 
+    var outMode = Arrays.asList(args).contains(bspFlag) ? OutFolderMode.BSP : OutFolderMode.REGULAR;
```

Comment:
This will trigger if any user-defined commands have `--bsp` as a flag, which may cause confusion. But maybe that's a rare enough occurrence we can ignore it for now

Reply:
I'm not sure that's true, IIRC intellij passes `-i --bsp` with `--bsp` in the second position

Reply:
I think whitelisting `--bsp` and `-i --bsp` as the first two args sounds reasonable. 

Alternatively, one thing we can do is if we identify a `--bsp` `-i` in the flags, we then run the argument parser in the launcher to see whether they correspond to the flags we care about. With Graal native launchers the cost may be low enough to be acceptable, but that will need to be measured